core/object.cpp

```diff
@@ -1366,7 +1366,10 @@ Array Object::_get_incoming_connections() const {
 void Object::get_signal_list(List<MethodInfo> *p_signals) const {
 
 	if (!script.is_null()) {
-		Ref<Script>(script)->get_script_signal_list(p_signals);
+		Ref<Script> scr = script;
+		if (scr.is_valid()) {
+			Ref<Script>(script)->get_script_signal_list(p_signals);
```

Comment:
I think you could dereference `scr` here directly?

Reply:
Yes I could! I'll update the PR.